lib/active_fedora/rdf/identifiable.rb

```diff
@@ -47,7 +47,7 @@ def resource_datastream(ds=nil)
     # @param [RDF::URI] uri URI that is being looked up.
     def from_uri(uri,_)
       begin
-        self.find(pid_from_subject(uri))
+        ActiveFedora::Base.find(pid_from_subject(uri))
```

Comment:
@jcoyne Example spec is here: https://github.com/projecthydra/active_fedora/blob/8.0-stable/spec/unit/rdf_resource_datastream_spec.rb#L197-L203.  DummyAsset model is not asserted by DummyResource.